ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts.py

```diff
@@ -1941,34 +1941,47 @@ def _overlay_vpn_ipv6(self) -> bool:
         )
 
     @cached_property
-    def overlay(self) -> dict:
+    def overlay_peering_address(self) -> str | None:
+        if not self.underlay_router:
+            return None
+
+        if self.overlay_routing_protocol_address_family == "ipv6":
+            return self.ipv6_router_id
+
+        return self.router_id
+
+    @cached_property
+    def overlay(self) -> dict | None:
         """
         Returns a dictionary of overlay parameters to configure on the node.
         """
-        # Set overlay.peering_address to use
-        if self.overlay_routing_protocol_address_family == "ipv6":
-            peering_address = self.ipv6_router_id
-        else:
-            peering_address = self.router_id
         # Set overlay.evpn_vxlan and overlay.evpn_mpls to differentiate between VXLAN and MPLS use cases.
         evpn_vxlan = self._overlay_evpn and self.evpn_encapsulation == "vxlan"
         evpn_mpls = self._overlay_evpn and self.evpn_encapsulation == "mpls"
         # Set dpath based on ipvpn_gateway parameters
         dpath = self._overlay_ipvpn_gateway and get(self._switch_data_combined, "ipvpn_gateway.enable_d_path", default=True)
 
-        return {
-            "peering_address": peering_address,
+        overlay_dict = {
             "ler": self._overlay_ler,
             "vtep": self._overlay_vtep,
             "evpn": self._overlay_evpn,
             "evpn_vxlan": evpn_vxlan,
-            "evpn_mpls": evpn_mpls,
             "vpn_ipv4": self._overlay_vpn_ipv4,
             "vpn_ipv6": self._overlay_vpn_ipv6,
             "ipvpn_gateway": self._overlay_ipvpn_gateway,
             "dpath": dpath,
         }
 
+        if self.underlay_router is True:
+            overlay_dict.update(
+                {
+                    "peering_address": self.overlay_peering_address,
+                    "evpn_mpls": evpn_mpls,
+                }
+            )
+
+        return overlay_dict
```

Comment:
I don't think you need this change. The entire structure of `overlay` has changed between the versions, so it is not possible to apply the same logic.

Reply:
We have kept this almost same, just updating the "overlay_dict" with latest devel changes, in the end we are not returning "None" (as it is in latest devel), we are returning the same dict what was there, just removing "peering_address" and "evpn_mpls" from it (as it is in latest devel).

Would you like us to change this back to how it was in 3.8?

Reply:
Yes I don't think these changes apply for the cherry pick.

Reply:
Ok.. In that case shall I remove [this](https://github.com/aristanetworks/ansible-avd/pull/3043/files#diff-ef9c24ca7d287ca54b0ee4167c4958d47845c503f9ac5db2cb228060144c4915R8) line as well?
otherwise we will get error similar to :

```
"Unexpected failure during module execution: 'overlay_routing_protocol_address_family: ipv6' is only supported in combination with 'underlay_ipv6: True' and 'underlay_rfc5549: True'"
```

Reply:
Fixed!